apex/transformer/parallel_state.py

```diff
@@ -247,7 +247,21 @@ def initialize_model_parallel(
             _POSITION_EMBEDDING_GLOBAL_RANKS = position_embedding_ranks
 
 
-def get_rank_info() -> Tuple[int, int, int]:
+def get_world_info() -> Tuple[int, int, int, int]:
+    """Returns a tuple of (data, tensor, pipeline, virtual pipeline)-parallel-world_size for logger."""
+    if model_parallel_is_initialized():
+        return (
+            get_data_parallel_world_size(),
+            get_tensor_model_parallel_world_size(),
+            get_pipeline_model_parallel_world_size(),
+            get_virtual_pipeline_model_parallel_world_size(),
+        )
+    else:
+        print("Warning: model and data parallel groups are not initialized")
```

Comment:
sorry for having confused you but by a warning I meant something like

```suggestion
        warnings.warn("model and data parallel groups are not initialized. Have you called `parallel_state.initialize_model_parallel`?")
```
which needs `import warnings`.

With this we can test this warning with `unittest` using e.g. https://docs.python.org/3/library/unittest.html#unittest.TestCase.assertWarnsRegex (needn't to access in this PR)